src/main/scala/com/ing/wbaa/gargoyle/proxy/provider/LineageProviderAtlas.scala

```diff
@@ -0,0 +1,143 @@
+        Seq(Classification("customer_PII"))))))
+  }
+
+  private def processEntities(
+      serverGuid: String,
+      bucketGuid: String,
+      fileGuid: String,
+      userSTS: String,
+      method: String,
+      inputGuid: String,
+      inputType: String,
+      outputGuid: String,
+      outputType: String,
+      timestamp: Long) = {
+    Entities(Seq(Ingestion(
+      "aws_cli_script",
+      userSTS,
+      IngestionAttributes(s"aws_cli_${timestamp}", s"aws_cli_${timestamp}", method, userSTS,
+        guidRef(serverGuid, "Server"),
+        List(guidRef(inputGuid, inputType)),
+        List(guidRef(outputGuid, outputType))))))
+  }
+
+  private def postEnities(userSTS: String, host: String, bucket: String, bucketObject: String, method: String, contentType: ContentType, timestamp: Long): Future[Option[LineagePostGuidResponse]] = {
+    for {
+      serverGuid <- postData(serverEntities(userSTS, host).toJson)
+      bucketGuid <- postData(bucketEntities(userSTS, bucket).toJson)
+      fileGuid <- method match {
+        case access if access == "read" =>
+          postData(
+            fileEntities(serverGuid, bucketGuid, bucketObject, userSTS, bucketGuid, "Bucket", serverGuid, "Server", contentType).toJson)
+        case access if access == "write" =>
+          postData(
+            fileEntities(serverGuid, bucketGuid, bucketObject, userSTS, serverGuid, "Server", bucketGuid, "Bucket", contentType).toJson)
+      }
+      processGuid <- method match {
+        case access if access == "read" =>
+          postData(
+            processEntities(serverGuid, bucketGuid, fileGuid, userSTS, method, bucketGuid, "Bucket", fileGuid, "DataFile", timestamp).toJson)
+        case access if access == "write" =>
+          postData(
+            processEntities(serverGuid, bucketGuid, fileGuid, userSTS, method, fileGuid, "DataFile", bucketGuid, "Bucket", timestamp).toJson)
+      }
+    } yield Some(LineagePostGuidResponse(serverGuid, bucketGuid, fileGuid, processGuid))
+  }
+
+  // file entity delete
+  // for now it is just deleting file entity and no related objects like eg. aws_cli_script, which uploaded or downloaded
+  // file to bucket. Once we delete aws_cli_script object we will lose track of whats has been deleted
+  // We need to come up with process of tracking file delete
+  private def deleteEntities(typeName: String, entityName: String): Future[Option[LineagePostGuidResponse]] = {
+    for {
+      entityGuid <- getEntityGUID(typeName, entityName)
+      _ <- deleteEntity(entityGuid)
+
+    } yield (Some(LineagePostGuidResponse("", "", entityGuid, "")))
+  }
+
+  def createLineageFromRequest(httpRequest: HttpRequest, userSTS: User): Future[Option[LineagePostGuidResponse]] = {
+
+    val timestamp = System.currentTimeMillis()
+    // createLineageFromRequest is used in RequestHandlerS3 where there is no notion of S3Request. It seems easier to
+    // repeat httpRequest parsing here for needed values
```

Comment:
If we move this function to the `ProxyService` class it would be nicer to use the s3Request (as the notion is available there).